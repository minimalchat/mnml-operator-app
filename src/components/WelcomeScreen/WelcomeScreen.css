.WelcomeScreen.screen {
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  background: #2c3e50;
}

.WelcomeScreen__header {
  width: 33%;
  min-width: 256px;
  margin: 18px 0 96px;
  display: flex;
  justify-content: center;
  align-items: center;
  font-family: Roboto, sans-serif;
  font-weight: 600;
  font-size: 13px;
  font-weight: 500;
  letter-spacing: 2px;
  text-transform: uppercase;
  color: white;
}

.WelcomeScreen__header .SVG__logo {
  height: 20px;
  margin-right: 10px;
  margin-bottom: 0;
  fill: rgba(255, 255, 255, 1.0);
}

.WelcomeScreen__body {
  width: 33%;
  min-width: 256px;
  max-width: 512px;
}

.WelcomeScreen__wrapper {
  position: relative;
}

.WelcomeScreen__wrapper input {
  width: 100%;
  height: 42px;
  box-sizing: border-box;
  padding: 14px;
  border-radius: 2px;
  border: 1px solid #ffffff;
}

.WelcomeScreen__wrapper input:focus {
  outline: none;
}

.WelcomeScreen__wrapper .Button {
  position: absolute;
  right: 3px;
  top: 3px;
}
