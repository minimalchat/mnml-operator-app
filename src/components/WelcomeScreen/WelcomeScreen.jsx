import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import BrandLogo from '../SVG/BrandLogo.jsx';
import Button from '../Button/Button.jsx';
import { updateSettings } from '../../store/Config';
import { toggleWelcomeScreen } from '../../store/UI';

import './WelcomeScreen.css';

class WelcomeScreen extends Component {
  static propTypes = {
    changeSettings: PropTypes.func.isRequired,
  }

  state = {
    connString: '',
  }

  connect = () => {
    const { changeSettings } = this.props;
    const { connString } = this.state;

    changeSettings({ apiServer: connString });
  }

  handleChange = (e) => {
    this.setState({
      [e.target.name]: e.target.value,
    });
  }

  render () {
    return (
      <div className="WelcomeScreen screen">
        <header className="WelcomeScreen__header">
          <BrandLogo />
          <span className="WelcomeScreen__appname">Operator</span>
        </header>
        <section className="WelcomeScreen__body">
          <form onSubmit={this.connect} className="WelcomeScreen__wrapper">
            <input
              type="text"
              name="connString"
              onChange={this.handleChange}
              value={this.state.connString}
              placeholder="https://api.server.com:1337"
            />
            <Button variant="send" type="submit">Connect</Button>
          </form>
        </section>
      </div>
    );
  }
}

const mapStateToProps = state => ({

});

const mapDispatchToProps = dispatch => ({
  changeSettings: newSettings => dispatch(updateSettings(newSettings)) &&
    dispatch(toggleWelcomeScreen(false)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(WelcomeScreen);
