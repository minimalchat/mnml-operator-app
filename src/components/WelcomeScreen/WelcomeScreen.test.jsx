import React from 'react';

import { shallow } from 'enzyme';

import WelcomeScreen from './WelcomeScreen.jsx';

const store = {
  subscribe: jest.fn(),
  dispatch: jest.fn(),
  getState: jest.fn(() => ({
    config: {
      apiServer: '',
    },
  })),
};

describe('WelcomeScreen', () => {
  it('matches snapshot', () => {
    const component = shallow(<WelcomeScreen store={store} />);

    expect(component).toMatchSnapshot();
  });
});
