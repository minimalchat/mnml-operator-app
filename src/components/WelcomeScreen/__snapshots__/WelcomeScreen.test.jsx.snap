// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeScreen matches snapshot 1`] = `
<WelcomeScreen
  changeSettings={[Function]}
  store={
    Object {
      "dispatch": [MockFunction],
      "getState": [MockFunction] {
        "calls": Array [
          Array [],
          Array [],
        ],
      },
      "subscribe": [MockFunction] {
        "calls": Array [
          Array [
            [Function],
          ],
        ],
      },
    }
  }
/>
`;
